Islandora-CLAW/islandora
# Issue 525 - POST request for FedoraResource entities

## What does this Pull Request do?
This PR is a first go at implementing: implements https://github.com/Islandora-CLAW/CLAW/issues/525 

Basically, it is should take a POST request to http://localhost:8000/fedora_resource/ and create a fedora_resource entity bundle.  The type of bundle is specified by the X-Islandora-Bundle header.  And the content or fields of the bundle is specified in the body as expanded json-ld.

## How should this be tested?
* This PR depends on https://github.com/Islandora-CLAW/islandora/pull/33.  While this PR is under review, you can pull that PR.  Then, merge pull this PR!  islandora.services.yml  is the only conflicting file.   
* Launch the POSTMAN
* Create a POST request with the sample body below and X-Islandora-Bundle header value rdf_source.  
```
{
    "@type": [
      "http://www.w3.org/ns/ldp#RDFSource",
      "http://www.w3.org/ns/ldp#Container"
    ],
    "http://purl.org/dc/elements/1.1/title": [
      {
        "@value": "This is a RDF Source"
      }
    ],
    "http://schema.org/dateCreated": [
      {
        "@value": "2017-02-16T20:26:06+00:00"
      }
    ],
    "http://schema.org/dateModified": [
      {
        "@value": "2017-02-16T20:26:06+00:00"
      }
    ]
}
```
* It should create it and return a Json message with the ID of the resource.


## Notes and Questions
* Checksum not implemented. 
* Please review how the rdf -> fields mapping is implemented.  There should be simpler ways!
* If two RDF triples are mapped to same field, which value should we take!
* Can we always assume expanded jsonld?  What are the supported formats?  Is there a library to do the jsonld conversions?
* Should most of the logic move into its own class?
